### PR TITLE
feat(azure): New check `storage_ensure_soft_delete_is_enabled`

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

This pr contains a the new Azure check `storage_ensure_soft_delete_is_enabled`

### Description

This check makes this task:

- Ensure Soft Delete is Enabled for Azure Containers and Blob Storage

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/storage/storage_ensure_soft_delete_is_enabled/storage_ensure_soft_delete_is_enabled.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    def execute(self) -> Check_Report_Azure:
+        findings = []
+        for subscription, storage_accounts in storage_client.storage_accounts.items():
+            for storage_account in storage_accounts:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.resource_name = storage_account.name
+                report.resource_id = storage_account.id
+                if (
+                    storage_account.blob_properties.container_delete_retention_policy.enabled
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/storage/storage_ensure_soft_delete_is_enabled/storage_ensure_soft_delete_is_enabled.py]: What happens if `storage_account.blob_properties` is None?

File: prowler/providers/azure/services/storage/storage_service.py [hunk 2]
```diff
@@ -42,11 +53,42 @@ def __get_storage_accounts__(self):
                 )
         return storage_accounts
 
+    def __get_blob_properties__(self):
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/storage/storage_service.py:56]: Can you add the test of this function in `storage_service_test.py`, pleae?

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

File: prowler/providers/azure/services/storage/storage_ensure_soft_delete_is_enabled/storage_ensure_soft_delete_is_enabled.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/services/storage/storage_ensure_soft_delete_is_enabled/storage_ensure_soft_delete_is_enabled.py]: True! I modify it to handle correctly.
Thanks for the review

File: prowler/providers/azure/services/storage/storage_service.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/services/storage/storage_service.py:56]: Done!

**Review by MrCloudSec — APPROVED** [review]
Great job!!

**Pull request merged**